pkg/handler/enqueue_annotation.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ EventHandler = &EnqueueRequestForAnnotation{}
+
+const (
+	// NamespaceAnnotation - annotation that will be used to get the primary resource namespace.
+	// E.g my-namespace
+	NamespaceAnnotation = "sigs.k8s.io/primary-resource-namespace"
+
+	// NameAnnotation - annotation that will be used to get the primary resource namespaced name.
+	// E.g my-resource-name
+	NameAnnotation = "sigs.k8s.io/primary-resource-name"
+
+	// TypeAnnotation - annotation that will be used to verify that the primary resource is the primary resource to use.
+	// E.g Pods.core
+	TypeAnnotation = "sigs.k8s.io/primary-resource-type"
+)
```

Comment:
Hi @joelanford, 

Also, I think is better we have 1 annotation for the namespace and other for the name. Otherwise, we need to verify if the "/" was used and make clear for the users that when it is cluster scoped the value should be "/name" which shows strange.  WDYT? Are you ok with?

Reply:
I don't think that's necessary. A slash character is not allowed in a valid DNS-1233 label (which is what names and namespaces must validate as), so IMO, the existing function used to split the string is sufficient.

Reply:
Hi @joelanford, 

It is fine . Done as you suggested.